src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -69,13 +71,17 @@
     /**
      * URL of JIRA for Jenkins access, like <tt>http://jira.codehaus.org/</tt>.
      * Mandatory. Normalized to end with '/'
+     * @deprecated use @{getUrl}
      */
+    @Deprecated
     public final URL url;
 
     /**
      * URL of JIRA for normal access, like <tt>http://jira.codehaus.org/</tt>.
      * Mandatory. Normalized to end with '/'
+     * @deprecated use @{getUrl}
      */
+    @Deprecated
```

Comment:
This is actually not Deprecated, see JiraChangeLogAnnotator :-)
(this is not obvious there I know... needs refactoring - alternativeUrl is used for links generated in HTML output, but not for Jenkins communication)
I'd say -> use getAlternativeUrl().
We should assume that getUrl always != null but alternativeUrl = getUrl if null